pkg/clients/postgresql/postgresql.go

```diff
@@ -3,32 +3,47 @@ package postgresql
 import (
 	"context"
 	"database/sql"
+	"fmt"
 
 	"github.com/crossplane-contrib/provider-sql/pkg/clients/xsql"
 
 	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
 	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
 )
 
+const (
+	sslModeKey = "sslmode"
+)
+
 type postgresDB struct {
 	dsn      string
 	endpoint string
 	port     string
+	sslmode  string
 }
 
 // New returns a new PostgreSQL database client.
 func New(creds map[string][]byte) xsql.DB {
 	// TODO(negz): Support alternative connection secret formats?
 	endpoint := string(creds[xpv1.ResourceCredentialsSecretEndpointKey])
 	port := string(creds[xpv1.ResourceCredentialsSecretPortKey])
+	sslmode := string(creds[sslModeKey])
+
+	dsn := fmt.Sprintf("host=%s port=%s user=%s password=%s",
```

Comment:
Switching to the preparsed space separated format avoids running `ParseURL` within `pq`, which contains some [escaping functionality](https://github.com/lib/pq/blob/b2901c7946b69f1e7226214f9760e31620499595/url.go#L46). 

My guess is that a password can contain special characters (quotes and backslashes according to the `pq` code) that could end up failing to parse without that escaping in place.

Would it be simpler to continue to use the old `dsn` format with a 'query string' argument (these are 'accrued' into the parsed string much like the other settings)? `?sslmode=...`

Reply:
I made the switch to a connection string because I passing query parameters didn't seem to work and [a comment](https://github.com/lib/pq/blob/b2901c7946b69f1e7226214f9760e31620499595/url.go#L16) for `ParseUrl` indicated the function to be there for backward compatibility. Since provider-sql wasn't supporting parameters it seemed save changing the format of the `dsn` string. 
I can retry using a connection URL, or is the `nurl.Parse(url)` already un-escaping special chars and is  `escaper.Replace()` putting them back?

Reply:
I have retested with the url format DSN and update the MR with this. Structure is code is equal, `sslmode` is only appended when provided.